src/plugins/vis_type_vega/public/expressions/line_vega_spec_fn.ts

```diff
@@ -0,0 +1,243 @@
+    },
+  };
+
+  // Get the valueAxes data and generate a map to easily fetch the different valueAxes data
+  const valueAxis = {};
+  visParams?.valueAxes.forEach((yAxis: { id: { toString: () => string | number } }) => {
+    // @ts-ignore
+    valueAxis[yAxis.id.toString()] = yAxis;
+  });
+
+  spec.layer = [] as any[];
+
+  if (datatable.rows.length > 0 && dimensions.x != null) {
+    const xAxisId = getXAxisId(dimensions, datatable.columns);
+    const xAxisTitle = cleanString(dimensions.x.label);
+    const startTime = new Date(dimensions.x.params.bounds.min).valueOf();
+    const endTime = new Date(dimensions.x.params.bounds.max).valueOf();
+    let skip = 0;
+    datatable.columns.forEach((column, index) => {
+      // Check if its not xAxis column data
+      if (column.meta?.aggConfigParams?.interval != null) {
+        skip++;
+      } else {
+        const currentSeriesParams = visParams.seriesParams[index - skip];
+        const currentValueAxis =
+          // @ts-ignore
+          valueAxis[currentSeriesParams.valueAxis.toString()];
+        let tooltip: Array<{ field: string; type: string; title: string }> = [];
+        if (visParams.addTooltip) {
+          tooltip = [
+            { field: xAxisId, type: 'temporal', title: xAxisTitle },
+            { field: column.id, type: 'quantitative', title: column.name },
+          ];
+        }
+        spec.layer.push({
+          mark: {
+            // Possible types are: line, area, histogram. The eligibility checker will
+            // prevent area and histogram (though area works in vega-lite)
+            type: currentSeriesParams.type,
+            // Possible types: linear, cardinal, step-after. All of these types work in vega-lite
+            interpolate: currentSeriesParams.interpolate,
+            // The possible values is any number, which matches what vega-lite supports
+            strokeWidth: currentSeriesParams.lineWidth,
+            // this corresponds to showing the dots in the visbuilder for each data point
+            point: currentSeriesParams.showCircles,
+          },
+          encoding: {
+            x: {
+              axis: {
+                title: xAxisTitle,
+                grid: visParams.grid.categoryLines,
+              },
+              field: xAxisId,
+              type: 'temporal',
+              scale: {
+                domain: [startTime, endTime],
+              },
+            },
+            y: {
+              axis: {
+                title: cleanString(currentValueAxis.title.text) || column.name,
+                grid: visParams.grid.valueAxis !== '',
+                orient: currentValueAxis.position,
+                labels: currentValueAxis.labels.show,
+                labelAngle: currentValueAxis.labels.rotate,
+              },
+              field: column.id,
+              type: 'quantitative',
+            },
+            tooltip,
+            color: {
+              datum: column.name,
+            },
```

Comment:
It might be helpful here to add a small comment about how/why this works, in that it will partition the different time series into their own colors for the legend

Reply:
yeah, it's one of those things that seems magical to anyone who has never worked with a vega-lite spec. Even then, still seems a little magical 😁

Reply:
Added comments to explain. This is to ensure none of the different columns end up having the same color.